x-pack/plugins/alerts/server/alerts_client/alerts_client.ts

```diff
@@ -619,6 +621,8 @@ export class AlertsClient {
       alertSavedObject = await this.unsecuredSavedObjectsClient.get<RawAlert>('alert', id);
     }
 
+    this.alertTypeRegistry.ensureAlertTypeEnabled(alertSavedObject.attributes.alertTypeId);
```

Comment:
I think all checks in this file that happen before authorization checks / audit logging should be moved to happen after.